apis/k8ssandra/v1alpha1/k8ssandracluster_types.go

```diff
@@ -313,6 +313,29 @@ type DatacenterOptions struct {
 	// a user-provided monitoring solution (at present, only support for Prometheus is available).
 	// +optional
 	Telemetry *telemetryapi.TelemetrySpec `json:"telemetry,omitempty"`
+
+	// Containers defines containers to be deployed in each Cassandra pod.
+	// K8ssandra-operator and cass-operator will create their own containers, which can be referenced here to control ordering.
+	// Example:
+	//  containers:
+	//  - name: server-system-logger
+	//  - name: custom-container
+	//    image: busybox
+	//  - name: cassandra
+	// Otherwise containers referenced here will be injected in first position
+	// +optional
+	Containers []corev1.Container `json:"containers,omitempty"`
+
+	// InitContainers defines init-containers to be deployed in each Cassandra pod.
+	// K8ssandra-operator and cass-operator will create their own init-containers, which can be referenced here to control ordering.
```

Comment:
How is the ordering handled with respect to init containers added by k8ssandra-operator?